src/main/java/org/apache/commons/fileupload/servlet/FileCleanerCleanup.java

```diff
@@ -83,7 +83,13 @@ public void contextInitialized(ServletContextEvent sce) {
      *     {@link #getFileCleaningTracker(ServletContext)}.
      */
     public void contextDestroyed(ServletContextEvent sce) {
-        getFileCleaningTracker(sce.getServletContext()).exitWhenFinished();
+        ServletContext context = sce.getServletContext();
+        if(context != null) {
```

Comment:
I don't think that this can happen. Can you describe your environment/use case some more?